### PR TITLE
Skip old coinbase block post Eunos

### PR DESCRIPTION
**Pull request by bvbfan**
/kind fix

#### What this PR does / why we need it:

Post Eunos coinbase spent less than pre-fork, so situation is old masternode can cause a bad block which will case an updated node to revert blocks till last checkpoint / anchor height. It can happen at any time post Eunos.

### CODE REVIEW

File: src/validation.cpp [hunk 1]
```diff
@@ -2730,9 +2724,7 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
                     if (!cache.HasICXSubmitEXTHTLCOpen(dfchtlc->offerTx))
                     {
                         CTokenAmount makerDeposit{DCT_ID{0}, offer->takerFee};
-                        auto res = cache.AddBalance(order->ownerAddress, makerDeposit);
-                        if (!res)
-                            LogPrintf("Can't refund makerDeposit back to owner: %s\n", res.msg);
+                        cache.AddBalance(order->ownerAddress, makerDeposit);
```
Comment (ShengguangXiao) [inline src/validation.cpp:2727]: Why skip the result check?

File: src/validation.cpp [hunk 2]
```diff
@@ -3541,6 +3523,16 @@ bool CChainState::ActivateBestChainStep(CValidationState& state, const CChainPar
                         // but it can be produced by outdated/malicious masternode
                         // so we should not shoutdown entire network, let's skip it
                         continue;
+                    } else if (reason == ValidationInvalidReason::CONSENSUS) {
+                        if (pindexConnect->nHeight >= chainparams.GetConsensus().EunosHeight) {
+                            auto strReason = state.GetRejectReason();
+                            // we have situaltion when old masternode will generate a block
```
Comment (ShengguangXiao) [inline src/validation.cpp:3529]: change situaltion -> situation

File: src/validation.cpp (hunk 1 above)
Comment (bvbfan) [reply src/validation.cpp:2727]: Since we can subtract we can add, it's safe. If add fails that's mean corrupt storage.

File: src/validation.cpp [hunk 3]
```diff
@@ -3523,6 +3523,16 @@ bool CChainState::ActivateBestChainStep(CValidationState& state, const CChainPar
                         // but it can be produced by outdated/malicious masternode
                         // so we should not shoutdown entire network, let's skip it
                         continue;
+                    } else if (reason == ValidationInvalidReason::CONSENSUS) {
+                        if (pindexConnect->nHeight >= chainparams.GetConsensus().EunosHeight) {
```
Comment (monstrobishi) [inline src/validation.cpp:3527]: Change this to `==`

File: src/validation.cpp (hunk 1 above)
Comment (Mixa84) [reply src/validation.cpp:2727]: I think we shouldn't do close order, offer, dfchtlc, exthtlc if Sub/AddBalance failed!

Comment (bvbfan) [reply src/validation.cpp:2727]: We should since if we not do it, order / offer and so on we have them in inconsistent state, they are not closed.

**Pull request merged**